packages/vite/src/node/plugins/worker.ts

```diff
@@ -28,6 +28,10 @@ interface WorkerCache {
 
 export type WorkerType = 'classic' | 'module' | 'ignore'
 
+export const workerOrSharedWorkerRE = /(?:\?|&)(worker|sharedworker)(?:&|$)/
+const workerFileRE = /(?:\?|&)worker_file&type=(\w+)(?:&|$)/
```

Comment:
For reference to others, we already depend on the order of query suffixes in other cases, like in html proxies. See https://github.com/vitejs/vite/blob/bdb826ca0a021c71d6a43d1f6c81fd3e906eea56/packages/vite/src/node/plugins/html.ts#L46.
So this should be ok.